File: Python3/695. Max Area of Island.py
class Solution:
    def maxAreaOfIsland(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        def countContinusArea(i, j):
            if i >= 0 and i < len(grid) and j >= 0 and j < len(grid[0]) and grid[i][j]:
                grid[i][j] = 0
                return 1 + countContinusArea(i-1, j) + countContinusArea(i+1, j) + countContinusArea(i, j-1) + countContinusArea(i, j+1)
            else:
                return 0

        maxArea = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                count = countContinusArea(i, j)
                maxArea = max(maxArea, count)
        
        return maxArea

File: Golang/695. Max Area of Island.go
func maxAreaOfIsland(grid [][]int) int {
    max := 0
    count := 0
    for i := range grid {
        for j := range grid[0] {
            count = countArea(&grid, i, j)
            if max < count {
                max = count
            }
        }
    }
    return max
}

func countArea(grid *[][]int, i, j int) int {
    if i >= 0 && i < len(*grid) && j >= 0 && j < len((*grid)[0]) && (*grid)[i][j] == 1 {
        (*grid)[i][j] = 0
        return 1 + countArea(grid, i-1, j) + countArea(grid, i+1, j) + countArea(grid, i, j-1) + countArea(grid, i, j+1)
    } else {
        return 0
    }
}
